src/basic/user-util.c

```diff
@@ -154,6 +153,17 @@ bool is_nologin_shell(const char *shell) {
                            "/usr/bin/true");
 }
 
+const char* default_root_shell(const char *root) {
+        const char *p = prefix_roota(root, DEFAULT_USER_SHELL);
+
+        /* We want to use the preferred shell, i.e. DEFAULT_USER_SHELL, which usually
+         * will be /bin/bash. But let's fall back to /bin/sh. */
+        if (access(p, X_OK) == 0)
```

Comment:
chase_symlinks() as usual...

(maybe add chase_symlinks_access() that combines chase_symlinks() and then uses faccess() on the final inode)

prefix_roota() should be considered